cms/djangoapps/contentstore/views/videos.py

```diff
@@ -835,3 +839,60 @@ def is_status_update_request(request_data):
     Returns True if `request_data` contains status update else False.
     """
     return any('status' in update for update in request_data)
+
+
+def download_youtube_video_thumbnail(youtube_id):
+    """
+    Download highest resoultion video thumbnail available from youtube.
+    """
+    thumbnail_content = thumbnail_content_type = None
+    # Download highest resoultion thumbnail available.
+    youtube_resolutions = ['maxresdefault', 'sddefault', 'hqdefault', '0']
+    for res in youtube_resolutions:
+        thumbnail_url = urljoin('https://img.youtube.com', '/vi/{youtube_id}/{res}.jpg'.format(
+            youtube_id=youtube_id, res=res
+        ))
+        response = requests.get(thumbnail_url)
+        if response.status_code == requests.codes.ok:   # pylint: disable=no-member
+            thumbnail_content = response.content
+            thumbnail_content_type = response.headers['content-type']
+            # If best available resolution is found, skip looking for lower resolutions.
+            break
+    return thumbnail_content, thumbnail_content_type
+
+
+def scrape_youtube_video_thumbnail(course_key_string, edx_video_id, youtube_id):
+    """
+    Scrapes video thumbnail from youtube.
+    """
+    thumbnail_content, thumbnail_content_type = download_youtube_video_thumbnail(youtube_id)
+    supported_content_types = {v: k for k, v in settings.VIDEO_IMAGE_SUPPORTED_FILE_FORMATS.iteritems()}
+    image_filename = '{youtube_id}{image_extention}'.format(
+        youtube_id=youtube_id,
+        image_extention=supported_content_types[thumbnail_content_type]
+    )
+    image_file = SimpleUploadedFile(image_filename, thumbnail_content, thumbnail_content_type)
+    error = validate_video_image(image_file, skip_aspect_ratio=True)
+    if error:
+        LOGGER.info(
+            'VIDEOS: Scraping youtube video thumbnail failed for edx_video_id [%s] in course [%s] with error: %s',
+            edx_video_id,
+            course_key_string,
+            error
+        )
+        return
+
+    update_video_image(edx_video_id, course_key_string, image_file, image_filename)
+    LOGGER.info(
+        'VIDEOS: Scraping youtube video thumbnail for edx_video_id [%s] in course [%s]', edx_video_id, course_key_string
+    )
+
+
+def scrape_youtube_thumbnails(videos):
+    """
+    Scrapes youtube thumbnails for given list of videos.
+    """
+    for course_id, edx_video_id, youtube_id in videos:
+        # Scrape when course video image does not exist for edx_video_id.
+        if not get_course_video_image_url(course_id, edx_video_id):
+            scrape_youtube_video_thumbnail(course_id, edx_video_id, youtube_id)
```

Comment:
This would be changed to something like:
```Python
for course_id, edx_video_id, youtube_id in videos:
        # Scrape when course video image does not exist for edx_video_id.
        if not get_course_video_image_url(course_id, edx_video_id):
                image_content, content_type = download_youtube_video_thumbnail(youtube_id)
                validate_and_update_video_image(course_id, edx_video_id,  image_content, content_type)
```